x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingResourceTracker.java

```diff
@@ -262,6 +277,35 @@ static void getMemoryAndProcessors(
             minNodes = Math.min(3, Math.max(minNodes, numberOfAllocations));
         }
 
+        // dummy autoscaling entity
+        //
+        // Is there enough capacity to accommodate the dummy entity?
+        // If there exists a node that can accommodate the dummy entity then return true (nothing to do),
+        // else return false and increment the memory and processor counts accordingly.
+        //
+        // We perform the calculation by identifying the least loaded node in terms of memory
+        // and determining if the addition of the dummy entity's memory & processor requirements could
+        // be accommodated on it.
+        //
+        // If the calculation returns false then treat the case as for a single trained model job
+        // that is already assigned, i.e. increment modelMemoryBytesSum and processorsSum appropriately.
+        //
+        if (checkIfDummyEntityCanBeAccommodatedOnLeastLoadedNode(
```

Comment:
The method name could be shorter.

```suggestion
        if (dummyEntityFitsOnLeastLoadedNode(
```